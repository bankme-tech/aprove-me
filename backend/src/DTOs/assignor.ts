import { IsNotEmpty, IsUUID, IsString, Length } from 'class-validator';

export class AssignorDto {
  @IsUUID()
  @IsNotEmpty()
  @IsString()
  id: string;

  @Length(5, 30)
  @IsNotEmpty()
  @IsString()
  document: string;

  @Length(5, 140)
  @IsNotEmpty()
  @IsString()
  email: string;

  @Length(5, 20)
  @IsNotEmpty()
  @IsString()
  phone: string;

  @Length(5, 140)
  @IsNotEmpty()
  @IsString()
  name: string;
}
