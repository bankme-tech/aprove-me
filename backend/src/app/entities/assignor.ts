import { Replace } from '@/utils/replace';
import { randomUUID } from 'crypto';

interface PayableProps {
  document: string;
  email: string;
  phone: string;
  name: string;
  createdAt: Date;
  updatedAt: Date;
  deletedAt?: Date;
}

interface PayablePropsReplaceble {
  createdAt?: Date;
  updatedAt?: Date;
}

export class Assignor {
  public _id: string;
  public props: PayableProps;

  constructor(
    props: Replace<PayableProps, PayablePropsReplaceble>,
    _id?: string,
  ) {
    this._id = _id ? _id : randomUUID();
    this.props = {
      ...props,
      createdAt: props.createdAt ?? new Date(),
      updatedAt: props.updatedAt ?? new Date(),
    };
  }
}
