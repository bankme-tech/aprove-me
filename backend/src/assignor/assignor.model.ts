import { Payable } from "@prisma/client";

export class Assignor {
    constructor(
        public id: number,
        public document: string,
        public email: string,
        public phone: string,
        public name: string,
        public payable: Payable[]
    ) { }
}