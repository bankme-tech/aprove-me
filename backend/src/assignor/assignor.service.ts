import { Injectable } from '@nestjs/common';
// import { AssignorRepository } from './repository.service';
import { AssignorRepository } from "./repository/repository.service";
import { Prisma, Assignor } from '@prisma/client';

@Injectable()
export class AssignorService {
  constructor(private readonly assignorRepository: AssignorRepository) {}

  async create(data: Prisma.AssignorCreateInput): Promise<Assignor> {
    return this.assignorRepository.create(data);
  }

  async findAll(): Promise<Assignor[]> {
    return this.assignorRepository.findAll();
  }

  async findOne(id: string): Promise<Assignor | null> {
    return this.assignorRepository.findOne(id);
  }

  async update(id: string, data: Prisma.AssignorUpdateInput): Promise<Assignor> {
    return this.assignorRepository.update(id, data);
  }

  async remove(id: string): Promise<Assignor> {
    return this.assignorRepository.delete(id);
  }
}
