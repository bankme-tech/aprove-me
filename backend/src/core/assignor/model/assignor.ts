import { Entity } from '@core/shared/model';
import { Notification } from '@core/shared/notification';
import { Document, Email, Name, Phone } from '@core/shared/value-objects';

export type AssignorProps = {
  id?: string;
  document: string;
  email: string;
  phone: string;
  name: string;
};

export class Assignor extends Entity<AssignorProps> {
  private _document: Document;
  private _email: Email;
  private _phone: Phone;
  private _name: Name;

  constructor(props: AssignorProps) {
    const notification = new Notification();
    super(props, notification);
    this._notification = notification;

    this._document = new Document(props.document, this._notification);
    this._email = new Email(props.email, this._notification);
    this._phone = new Phone(props.phone, this._notification);
    this._name = new Name(props.name, this._notification);
  }

  static create(props: AssignorProps): Assignor {
    return new Assignor(props);
  }

  public update(props: Partial<AssignorProps>): void {
    this._document = new Document(props.document, this._notification);
    this._email = new Email(props.email, this._notification);
    this._phone = new Phone(props.phone, this._notification);
    this._name = new Name(props.name, this._notification);
  }

  get id() {
    return this._id.getValue();
  }

  get document() {
    return this._document.getValue();
  }

  get email() {
    return this._email.getValue();
  }

  get phone() {
    return this._phone.getValue();
  }

  get name() {
    return this._name.getValue();
  }

  get notifications() {
    return this.getNotifications();
  }

  get containNotifications() {
    return this.hasNotifications();
  }
}
