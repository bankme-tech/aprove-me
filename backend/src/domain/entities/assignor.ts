import { Assignor as PrismaAssignor } from '@prisma/client';
import { randomUUID } from 'crypto';

export class Assignor implements PrismaAssignor {
  id: string;
  created_at: Date;
  document: string;
  email: string;
  name: string;
  phone: string;
  deleted_at: Date | null;
  updated_at: Date;

  static create(
    props: Omit<Assignor, 'id' | 'created_at' | 'updated_at' | 'deleted_at'>,
    id?: string,
  ) {
    const assignor = new Assignor();
    assignor.id = id ?? randomUUID();
    assignor.created_at = new Date();
    assignor.updated_at = new Date();
    assignor.deleted_at = null;
    assignor.document = props.document;
    assignor.email = props.email;
    assignor.name = props.name;
    assignor.phone = props.phone;

    return assignor;
  }
}
