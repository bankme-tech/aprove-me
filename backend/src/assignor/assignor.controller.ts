import {
  Controller,
  Get,
  Post,
  Body,
  Patch,
  Param,
  Delete,
  UseGuards,
  HttpCode,
} from '@nestjs/common';
import { AssignorService } from './assignor.service';
import { CreateAssignorDto } from './dto/create-assignor.dto';
import { UpdateAssignorDto } from './dto/update-assignor.dto';
import { AuthGuard } from '../auth/auth.guard';

@UseGuards(AuthGuard)
@Controller()
export class AssignorController {
  constructor(private readonly assignorService: AssignorService) {}

  @Post()
  async create(@Body() createAssignorDto: CreateAssignorDto) {
    return this.assignorService.create(createAssignorDto);
  }

  @Get()
  async findAll() {
    return this.assignorService.findAll();
  }

  @Get(':id')
  async findOne(@Param('id') id: string) {
    return this.assignorService.findOne(id);
  }

  @Patch(':id')
  async update(@Param('id') id: string, @Body() updateAssignorDto: UpdateAssignorDto) {
    return this.assignorService.update(id, updateAssignorDto);
  }

  @HttpCode(204)
  @Delete(':id')
  async remove(@Param('id') id: string) {
    await this.assignorService.remove(id);
    return { message: 'Assignor deleted' };
  }
}
