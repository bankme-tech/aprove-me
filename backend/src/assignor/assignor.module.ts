import { Module } from '@nestjs/common';
import { AssignorService } from './assignor.service';
import { AssignorController } from './assignor.controller';
import { AssignorRepository} from './repository/repository.service';
import { PrismaService } from '../prisma/prisma.service';
import { SessionModule } from 'src/auth/session/session-manager.module';

@Module({
  imports: [SessionModule],
  controllers: [AssignorController],
  providers: [AssignorService, AssignorRepository, PrismaService],
  exports: [AssignorService],
})
export class AssignorModule {}
