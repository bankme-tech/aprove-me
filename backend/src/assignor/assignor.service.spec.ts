import { Test, TestingModule } from "@nestjs/testing";
import { AssignorService } from "./assignor.service";
import { PrismaService } from "../prisma.service";
import { Assignor } from "./assignor.model";
import { mockDeep, DeepMockProxy } from 'jest-mock-extended';
import { PrismaClient } from "@prisma/client";

const assignor = new Assignor();
assignor.id = '123';
assignor.document = '12345678900';
assignor.email = 'test@test.com';
assignor.phone = '5521999999999';
assignor.name = 'Test Assignor';

const assignors: Assignor[] = [assignor]

describe("AssignorService", () => {
    let assignorService: AssignorService;
    let prismaService: PrismaService;

    beforeEach(async () => {
        const module: TestingModule = await Test.createTestingModule({
            providers: [AssignorService, PrismaService]
        })
            .overrideProvider(PrismaService)
            .useValue(mockDeep(PrismaClient))
            .compile();

        assignorService = module.get(AssignorService);
        prismaService = module.get(PrismaService);

        prismaService.assignor.findMany = jest.fn().mockResolvedValue(assignors);
        prismaService.assignor.findFirst = jest.fn().mockResolvedValue(assignor);
        prismaService.assignor.create = jest.fn().mockResolvedValue(assignor);
        prismaService.assignor.update = jest.fn().mockResolvedValue(assignor);
        prismaService.assignor.delete = jest.fn().mockResolvedValue(assignor);
    });

    it("should be defined", () => {
        expect(assignorService).toBeDefined();
        expect(prismaService).toBeDefined();
    });

    describe("getAllAssignors", () => {
        it("should return a list of assignors successfully", async () => {
            const result = await assignorService.getAllAssignors();

            expect(result).toEqual(assignors);
        });
    });

    describe("getAssignorById", () => {
        it("should return one assignor successfully with the same param id sent", async () => {
            const id = '123';

            const result = await assignorService.deleteAssignorById(id);

            expect(result.id).toBe(id);
        });
    });

    describe("createAssignor", () => {
        it("should return one assignor successfully with the same data sent with id", async () => {
            const newAssignor = new Assignor();
            newAssignor.document = '12345678900';
            newAssignor.email = 'test@test.com';
            newAssignor.phone = '5521999999999';
            newAssignor.name = 'Test Assignor';

            const result = await assignorService.createAssignor(newAssignor);

            expect(result.id).not.toBe(null);
            expect(result.document).toBe(newAssignor.document);
            expect(result.email).toBe(newAssignor.email);
            expect(result.phone).toBe(newAssignor.phone);
            expect(result.name).toBe(newAssignor.name);
        });
    });

    describe("updateAssignor", () => {
        it("should return one assignor successfully with the same data sent", async () => {
            const result = await assignorService.updateAssignor(assignor);

            expect(result).toEqual(assignor);
        });
    });

    describe("deleteAssignorById", () => {
        it("should return one assignor sucessfully with the same id sent", async () => {
            const id = '123';

            const result = await assignorService.deleteAssignorById(id);

            expect(result.id).toBe(id);
        });
    });
});