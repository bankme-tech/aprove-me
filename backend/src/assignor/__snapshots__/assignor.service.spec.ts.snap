// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic CRUD operation (assignor) should delete a payable 1`] = `
{
  "id": "f880ae47-39fe-4ea7-8d3d-210bbd179b63",
  "name": "Assignor Name",
}
`;

exports[`basic CRUD operation (assignor) should return all assignor 1`] = `
[
  {
    "name": "Henrique",
  },
  {
    "name": "Mauler",
  },
]
`;

exports[`basic CRUD operation (assignor) should return an assignor by id 1`] = `
{
  "id": "1",
  "name": "Frederico",
}
`;

exports[`basic CRUD operation (assignor) should update an payable by id 1`] = `
{
  "amount": 1000,
  "assignId": "1",
  "description": "An updated description",
  "id": "1",
}
`;
